### ISSUE TITLE
`CompactStartElement` appears to incorrectly classify attributes as default

### ISSUE THREAD

**Issue by jtuc** [FasterXML/woodstox#43]
When reading and writing XML via `WstxInputFactory` and `DefaultEventAllocator`, attributes appear to be classed as default/unspecified incorrectly.

This appears to be caused by `CompactStartElement.constructAttr`; this method passes `isDef` (presumably "is default"?) to `AttributeEventImpl`'s `wasSpecified` parameter directly. If an attribute is default, shouldn't that mean it wasn't specified?

Please let me know if I'm misinterpreting this behaviour.

Test case:
```java
import java.io.ByteArrayInputStream;
import java.io.StringWriter;
import javax.xml.stream.XMLEventFactory;
import javax.xml.stream.XMLStreamException;
import javax.xml.stream.XMLStreamReader;
import javax.xml.stream.events.StartElement;
import javax.xml.stream.events.XMLEvent;

import com.ctc.wstx.evt.DefaultEventAllocator;
import com.ctc.wstx.stax.WstxEventFactory;
import com.ctc.wstx.stax.WstxInputFactory;

public class Main {

    public static void main(String[] args) throws XMLStreamException {
        String xml = "<?xml version=\"1.0\" ?><a b=\"c\"></a>";
        WstxInputFactory inputFactory = new WstxInputFactory();
        XMLStreamReader stream = inputFactory.createXMLStreamReader(new ByteArrayInputStream(xml.getBytes()));
        DefaultEventAllocator allocator = DefaultEventAllocator.getDefaultInstance();
        XMLEventFactory eventFactory = new WstxEventFactory();
        StringWriter writer = new StringWriter();
        while(stream.hasNext()) {
            XMLEvent event = allocator.allocate(stream);
            if(event.isStartElement()) {
                // Force parsing of attributes
                StartElement start = event.asStartElement();
                event = eventFactory.createStartElement(start.getName(), start.getAttributes(), start.getNamespaces());
            }
            event.writeAsEncodedUnicode(writer);
            stream.next();
        }
        System.out.println("Expected: " + xml);
        System.out.println("Actual:   " + writer.toString());
    }
}
```

**Comment by cowtowncoder** [comment]
Ok. One suggestion: instead of printing out textual output, it'd be good to test directly parts that seem incorrect, if possible, with assertions.

Also: which version of Woodstox is this with? (just to make sure you use the latest patch(ed) version).

**Comment by jtuc** [comment]
Sorry, I left in a fair bit of boilerplate just to make sure I was using things correctly.
I'm using Woodstox 5.0.3.

While trying to get a more minimal test case, I discovered that this bug actually seems to be specific to fetching attributes via the iterator:

```java
import com.ctc.wstx.evt.DefaultEventAllocator;
import com.ctc.wstx.stax.WstxInputFactory;
import org.junit.Before;
import org.junit.Test;

import javax.xml.namespace.QName;
import javax.xml.stream.XMLStreamException;
import javax.xml.stream.XMLStreamReader;
import javax.xml.stream.events.Attribute;
import javax.xml.stream.events.StartElement;
import java.io.StringReader;

import static org.junit.Assert.assertTrue;

public class AttributeTest {
    private StartElement element;

    @Before
    public void setupElement() throws XMLStreamException {
        String xml = "<a b=\"c\"/>";
        WstxInputFactory inputFactory = new WstxInputFactory();
        XMLStreamReader reader = inputFactory.createXMLStreamReader(new StringReader(xml));
        DefaultEventAllocator allocator = DefaultEventAllocator.getDefaultInstance();
        reader.next();
        element = allocator.allocate(reader).asStartElement();
    }

    @Test
    public void whenFetchedByName_attributeIsSpecified() {
        Attribute attribute = element.getAttributeByName(QName.valueOf("b"));
        assertTrue(attribute.isSpecified()); // Passes
    }

    @Test
    public void whenFetchedByIterator_attributeIsSpecified() {
        Attribute attribute = (Attribute) element.getAttributes().next();
        assertTrue(attribute.isSpecified()); // Fails
    }
}
```

**Comment by cowtowncoder** [comment]
@jtuc Thank you. This is helpful. I have a feeling like something like this was reported earlier... perhaps an earlier fix was incomplete.

**Comment by cowtowncoder** [comment]
Yup. That one negation in there was wrong... I suspect it must have been result of an earlier refactoring in which naming was changed from "is specified" to "is default", but logic not. Possibly back when basic event implementations were added in `stax2-api` package.

At any rate I fixed this, and fix will be in 5.1.0. I'll see if there might be any other easy fixes to make.
